commands/addResource.ts

```diff
@@ -1,56 +1,270 @@
+  const tags = await readTags();
+  const tagsOptions = tags.map(tag => ({ label: tag.name, value: tag.id }));
+  const tagsRow = new MessageActionRow().addComponents(
+    new MessageSelectMenu()
+      .setCustomId('tags')
+      .setPlaceholder('Select tags')
+      .setMaxValues(Math.min(tagsOptions.length, 25))
+      .addOptions(tagsOptions),
+  );
+
+  const blockchain = await readBlockchain();
+  const bcOptions = blockchain.map(bc => ({ label: bc.name, value: bc.id }));
+  const bcRow = new MessageActionRow().addComponents(
+    new MessageSelectMenu()
+      .setCustomId('blockchain')
+      .setPlaceholder('Select blockchain')
+      .setMaxValues(Math.min(bcOptions.length, 25))
+      .addOptions(bcOptions),
+  );
+
+  const categories = await readCategory();
+  const categoryOptions = categories.map(category => ({ label: category.name, value: category.id }));
+  const categoryRow = new MessageActionRow().addComponents(
+    new MessageSelectMenu()
+      .setCustomId('category')
+      .setPlaceholder('Select categories')
+      .setMaxValues(Math.min(categoryOptions.length, 25))
+      .addOptions(categoryOptions),
+  );
+
+  const authors = await readAuthors();
+  const authorOptions = authors.map(author => ({ label: author.name, value: author.id }));
+  const authorRow = new MessageActionRow().addComponents(
+    new MessageSelectMenu()
+      .setCustomId('author')
+      .setPlaceholder('Select author')
+      .addOptions(authorOptions),
+  );
+
+  const selectionRows = [authorRow, bcRow, categoryRow, tagsRow];
+
+  await interaction.editReply({
+    embeds: [resourceEmbed],
+    content: 'Tell me about the resource',
+    components: selectionRows,
+  });
+
+  const collector = interaction.channel?.createMessageComponentCollector({
+    maxComponents: 4,
+    time: 60_000,
+    componentType: 'SELECT_MENU',
+  });
+
+  collector?.on('collect', (menuInteraction) => {
+    switch (menuInteraction.customId) {
+      case 'category': {
+        resource.category = menuInteraction.values.map(v => {
+          const lookupItem = categories.find((value) => value.id === v);
+          return lookupItem ?? { name: 'Unknown', id: v };
+        });
+        break;
+      }
+      case 'tags': {
+        resource.tags = menuInteraction.values.map(v => {
+          const lookupItem = tags.find((value) => value.id === v);
+          return lookupItem ?? { name: 'Unknown', id: v };
+        });
+        break;
+      }
+      case 'blockchain': {
+        resource.blockchain = menuInteraction.values.map(v => {
+          const lookupItem = blockchain.find((value) => value.id === v);
+          return lookupItem ?? { name: 'Unknown', id: v };
+        });
+        break;
+      }
+      case 'author': {
+        resource.author = menuInteraction.values.map(v => {
+          const lookupItem = authors.find((value) => value.id === v);
+          return lookupItem ?? { name: 'Unknown', id: v };
+        })[0];
```

Comment:
None of the others have this index at the end. I'm a bit confused what this is doing? My understanding was that author only support one entry.

Reply:
The return from the menuInteraction is always an array of strings. All of the other select menus are multi select and we want arrays of selected items. In the case of the Author we only allow a single select, so grab the first and _only_ entry in the array.